src/wix/test/WixToolsetTest.CoreIntegration/VersionFixture.cs

```diff
@@ -12,6 +12,66 @@ namespace WixToolsetTest.CoreIntegration
 
     public class VersionFixture
     {
+        [Fact]
+        public void CanBuildMsiWithPrefixedVersion()
+        {
+            var folder = TestData.Get(@"TestData");
+
+            using (var fs = new DisposableFileSystem())
+            {
+                var baseFolder = fs.GetFolder();
+                var intermediateFolder = Path.Combine(baseFolder, "obj");
+                var msiPath = Path.Combine(baseFolder, "bin", "test1.msi");
+
+                var result = WixRunner.Execute(new[]
+                {
+                    "build",
+                    Path.Combine(folder, "Version", "PackageWithReplaceableVersion.wxs"),
+                    "-bindpath", Path.Combine(folder, "SingleFile", "data"),
+                    "-intermediateFolder", intermediateFolder,
+                    "-d", "Version=v4.3.2.1",
```

Comment:
In order to close the issue, this needs to be a bind variable instead of a preprocessor variable.